data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/cnec/FlowCnecCreator.java

```diff
@@ -0,0 +1,321 @@
+import com.farao_community.farao.data.crac_api.Contingency;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_api.Instant;
+import com.farao_community.farao.data.crac_api.cnec.FlowCnecAdder;
+import com.farao_community.farao.data.crac_api.cnec.Side;
+import com.farao_community.farao.data.crac_api.threshold.BranchThresholdAdder;
+import com.farao_community.farao.data.crac_creation.creator.api.ImportStatus;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileConstants;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileCracCreationContext;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileElementaryCreationContext;
+import com.powsybl.iidm.network.*;
+import com.powsybl.triplestore.api.PropertyBag;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.*;
+
+public class FlowCnecCreator extends AbstractCnecCreator {
+
+    private enum FlowCnecDefinitionMode {
+        CONDUCTING_EQUIPMENT,
+        OPERATIONAL_LIMIT,
+        WRONG_DEFINITION;
+    }
+
+    private final String conductingEquipment;
+    private final Set<Side> defaultMonitoredSides;
+
+    public FlowCnecCreator(Crac crac, Network network, String assessedElementId, String nativeAssessedElementName, String assessedElementOperator, boolean inBaseCase, PropertyBag currentLimitPropertyBag, String conductingEquipment, List<Contingency> linkedContingencies, Set<CsaProfileElementaryCreationContext> csaProfileCnecCreationContexts, CsaProfileCracCreationContext cracCreationContext, Set<Side> defaultMonitoredSides, String rejectedLinksAssessedElementContingency) {
+        super(crac, network, assessedElementId, nativeAssessedElementName, assessedElementOperator, inBaseCase, currentLimitPropertyBag, linkedContingencies, csaProfileCnecCreationContexts, cracCreationContext, rejectedLinksAssessedElementContingency);
+        this.conductingEquipment = conductingEquipment;
+        this.defaultMonitoredSides = defaultMonitoredSides;
+    }
+
+    public void addFlowCnecs() {
+        FlowCnecDefinitionMode definitionMode = getCnecDefinitionMode();
+        if (definitionMode == FlowCnecDefinitionMode.WRONG_DEFINITION) {
+            return;
+        }
+
+        String networkElementId = definitionMode == FlowCnecDefinitionMode.CONDUCTING_EQUIPMENT ? conductingEquipment : operationalLimitPropertyBag.getId(CsaProfileConstants.REQUEST_OPERATIONAL_LIMIT_TERMINAL);
+        Identifiable<?> branch = getFlowCnecBranch(networkElementId);
+        if (branch == null) {
+            return;
+        }
+
+        Pair<EnumMap<Branch.Side, Double>, Map<Integer, EnumMap<Branch.Side, Double>>> thresholds = definitionMode == FlowCnecDefinitionMode.CONDUCTING_EQUIPMENT ? getPermanentAndTemporaryLimitsOfBranch((Branch<?>) branch) : getPermanentAndTemporaryLimitsOfOperationalLimit(branch, networkElementId);
```

Comment:
In my opinion, this Pair type isn't very useful, it would be more readable if you use only the Map on the right of the pair, with the key Integer.MAX_VALUE for patl for example. You only have to extract this from the Map on line 50 and suppress the key from the map on line 51.
Alternatively, you can have a method for patl, and one for tatl, with definitionMode in argument, to cut algorithm in another way and avoid this Pair encapsulation.